File: design/icones/icones.js
/* Genere par outils/sous-ensembler-icones.mjs — ne pas modifier a la main.
   Table nom d'icone Material Symbols -> caractere dans la police
   sous-ensemblee. Ajouter une icone : la lister dans
   outils/icones-utilisees.txt puis relancer le script. */

export const ICONES = {
  today: '\ue8df',
  checkroom: '\uf19e',
  group: '\ue7ef',
  photo_library: '\ue413',
  storefront: '\uea12',
  arrow_back: '\ue5c4',
  close: '\ue14c',
  search: '\ue8b6',
  more_vert: '\ue5d4',
  contrast: '\ueb37',
  menu: '\ue5d2',
  chevron_right: '\ue409',
  expand_more: '\ue5cf',
  arrow_forward: '\ue5c8',
  open_in_new: '\ue895',
  add: '\ue145',
  edit: '\ue150',
  delete: '\ue872',
  check: '\ue5ca',
  check_circle: '\ue86c',
  save: '\ue161',
  content_copy: '\ue14d',
  download: '\ue171',
  share: '\ue80d',
  send: '\ue163',
  call: '\ue0b0',
  person_add: '\ue7fe',
  photo_camera: '\ue3b0',
  add_photo_alternate: '\ue43e',
  image: '\ue251',
  calendar_month: '\uebcc',
  filter_list: '\ue152',
  visibility: '\ue417',
  content_cut: '\ue14e',
  iron: '\ue583',
  inventory_2: '\ue1a1',
  schedule: '\ue192',
  warning: '\ue002',
  priority_high: '\ue645',
  error: '\ue000',
  info: '\ue88e',
  notifications: '\ue7f4',
  notifications_active: '\ue7f7',
  straighten: '\ue41c',
  payments: '\uef63',
  person: '\ue7fd',
  upload: '\ue2c6',
  receipt_long: '\uef6e',
  sell: '\ue54e',
  history: '\ue28e',
  picture_as_pdf: '\ue415',
  celebration: '\uea65',
  cloud_off: '\ue2c1',
  wifi_off: '\ue648',
  backup: '\ue864',
  settings_backup_restore: '\ue8ba',
  sentiment_satisfied: '\ue0ed',
};

/* Rend une icone. Decorative par defaut : le texte a cote dit deja la
   meme chose. Passer { titre } uniquement quand l'icone est seule
   porteuse de sens. */
export function icone(nom, { taille = 24, classe = '', pleine = false, titre = '' } = {}) {
  const glyphe = ICONES[nom];
  if (!glyphe) throw new Error(`Icone absente du sous-ensemble : ${nom}`);
  const classes = ['material-symbols-rounded', `md-icon-${taille}`];
  if (pleine) classes.push('md-icon-filled');
  if (classe) classes.push(classe);
  const acces = titre ? `role="img" aria-label="${titre}"` : 'aria-hidden="true"';
  return `<span class="${classes.join(' ')}" ${acces}>${glyphe}</span>`;
}

File: android/app/src/main/kotlin/com/essama/dresscode/charte/Icones.kt
package com.essama.dresscode.charte

import androidx.compose.runtime.Immutable

/*
 * Genere par outils/generer-icones-android.mjs — ne pas modifier a la main.
 *
 * Les icones sont les Material Symbols Rounded du kit, sous-ensemblees
 * aux 57 reellement utilisees. On les adresse par leur point de
 * code et non par leur ligature : le sous-ensemblage elague les regles
 * GSUB de la police, et une ligature perdue afficherait le mot
 * « straighten » a la place du glyphe.
 *
 * Ajouter une icone : la lister dans outils/icones-utilisees.txt,
 * relancer outils/sous-ensembler-icones.mjs puis
 * outils/generer-polices-android.mjs et ce script.
 */

@Immutable
@JvmInline
value class Icone(val glyphe: String)

object Icones {
    val Today = Icone("\ue8df")
    val Checkroom = Icone("\uf19e")
    val Group = Icone("\ue7ef")
    val PhotoLibrary = Icone("\ue413")
    val Storefront = Icone("\uea12")
    val ArrowBack = Icone("\ue5c4")
    val Close = Icone("\ue14c")
    val Search = Icone("\ue8b6")
    val MoreVert = Icone("\ue5d4")
    val Contrast = Icone("\ueb37")
    val Menu = Icone("\ue5d2")
    val ChevronRight = Icone("\ue409")
    val ExpandMore = Icone("\ue5cf")
    val ArrowForward = Icone("\ue5c8")
    val OpenInNew = Icone("\ue895")
    val Add = Icone("\ue145")
    val Edit = Icone("\ue150")
    val Delete = Icone("\ue872")
    val Check = Icone("\ue5ca")
    val CheckCircle = Icone("\ue86c")
    val Save = Icone("\ue161")
    val ContentCopy = Icone("\ue14d")
    val Download = Icone("\ue171")
    val Share = Icone("\ue80d")
    val Send = Icone("\ue163")
    val Call = Icone("\ue0b0")
    val PersonAdd = Icone("\ue7fe")
    val PhotoCamera = Icone("\ue3b0")
    val AddPhotoAlternate = Icone("\ue43e")
    val Image = Icone("\ue251")
    val CalendarMonth = Icone("\uebcc")
    val FilterList = Icone("\ue152")
    val Visibility = Icone("\ue417")
    val ContentCut = Icone("\ue14e")
    val Iron = Icone("\ue583")
    val Inventory2 = Icone("\ue1a1")
    val Schedule = Icone("\ue192")
    val Warning = Icone("\ue002")
    val PriorityHigh = Icone("\ue645")
    val Error = Icone("\ue000")
    val Info = Icone("\ue88e")
    val Notifications = Icone("\ue7f4")
    val NotificationsActive = Icone("\ue7f7")
    val Straighten = Icone("\ue41c")
    val Payments = Icone("\uef63")
    val Person = Icone("\ue7fd")
    val Upload = Icone("\ue2c6")
    val ReceiptLong = Icone("\uef6e")
    val Sell = Icone("\ue54e")
    val History = Icone("\ue28e")
    val PictureAsPdf = Icone("\ue415")
    val Celebration = Icone("\uea65")
    val CloudOff = Icone("\ue2c1")
    val WifiOff = Icone("\ue648")
    val Backup = Icone("\ue864")
    val SettingsBackupRestore = Icone("\ue8ba")
    val SentimentSatisfied = Icone("\ue0ed")
}
